potpie/daemon/process/pidfile.py

```
@@ -25,14 +25,19 @@ def _pid_alive(pid: int) -> bool:
         try:
-            existing = int(path.read_text().strip())
-        except ValueError:
-            existing = -1
-        if existing > 0 and _pid_alive(existing):
-            raise AlreadyRunning(f"daemon already running (pid={existing})")
-    _write_private_text(path, f"{pid}\n")
+            fd = os.open(path, os.O_WRONLY | os.O_CREAT | os.O_EXCL, 0o600)
+        except FileExistsError:
+            existing = read_pid_file(path)
+            if existing is not None and existing > 0 and _pid_alive(existing):
+                raise AlreadyRunning(f"daemon already running (pid={existing})")
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)